rust/src/parser.rs

```diff
@@ -26,6 +26,37 @@ use crate::applayer;
 use std::os::raw::{c_void,c_char,c_int};
 use crate::applayer::{AppLayerGetTxIterTuple};
 
+#[repr(C)]
+pub struct AppLayerReturn {
```

Comment:
@jasonish any thoughts on naming here? Maybe `AppLayerResult` would be more consistent with `OutputInitResult` we use already in other places.

Reply:
Yes, I think Result is a better name.  It just doesn't map the well used Result types in Rust which is like Result<TheReturnValue, TheError>... But I don't think I'd worry about that too much.

Reply:
Ok will change.